packages/x-adapter/src/endpoint-adapter.factory.ts

```diff
@@ -29,16 +30,17 @@ export const endpointAdapterFactory: EndpointAdapterFactory = <Request, Response
       endpoint: rawEndpoint,
       httpClient = fetchHttpClient,
       requestMapper = identityMapper,
-      responseMapper = identityMapper
+      responseMapper = identityMapper,
+      defaultRequestOptions = {}
     } = options;
 
     const endpoint = getEndpoint(rawEndpoint ?? requestEndpoint, request);
     const requestParameters = requestMapper(request, { endpoint });
 
-    return httpClient<Response>(endpoint, {
-      ...requestOptions,
-      parameters: requestParameters
-    }).then(response => responseMapper(response, { endpoint, requestParameters }));
+    return httpClient<Response>(
+      endpoint,
+      deepMerge({}, defaultRequestOptions, requestOptions, { parameters: requestParameters })
```

Comment:
I've moved the `parameters` inside the `deepMerge` so `undefined` values don't override the `defaultRequestOptions`.